### ISSUE TITLE
Annotation processor crash when @Generated unavailable

### ISSUE THREAD

**Issue by pyricau** [google/dagger#334]
I forgot to add a provided dependency on `javax.annotation:jsr250-api:1.0`, and then I got this error message:

```
Caused by: java.lang.IllegalArgumentException: unused arguments: expected 0, received 2
    at com.squareup.javapoet.Util.checkArgument(Util.java:62)
    at com.squareup.javapoet.CodeBlock$Builder.add(CodeBlock.java:180)
    at com.squareup.javapoet.JavaFile$Builder.addFileComment(JavaFile.java:217)
    at dagger2.internal.codegen.JavaPoetSourceFileGenerator.buildJavaFile(JavaPoetSourceFileGenerator.java:109)
    at dagger2.internal.codegen.JavaPoetSourceFileGenerator.generate(JavaPoetSourceFileGenerator.java:74)
    at dagger2.internal.codegen.ModuleProcessingStep.process(ModuleProcessingStep.java:125)
```

Looking at the code, the culprit seems to be in [JavaPoetSourceFileGenerator.buildJavaFile()](https://github.com/google/dagger/blame/master/compiler/src/main/java/dagger/internal/codegen/JavaPoetSourceFileGenerator.java#L105-L108):

```java
    if (!generatedAnnotationAvailable) {
      javaFileBuilder.addFileComment(
          "Generated by %s (%s).", ComponentProcessor.class.getName(), GENERATED_COMMENTS);
    }
```

**Comment by ronshapiro** [comment]
Shoot, those should be $S not %s. I'll have a fix tomorrow morning, thanks
for reporting.

**Comment by gk5885** [comment]
@ronshapiro, is this fixed then?

**Comment by ronshapiro** [comment]
Yup, in 6cc3c2335846f67515d29a44f30f763468ff6e52 and released in 2.2
